compiler/src/iree/compiler/Dialect/Flow/IR/FlowOpFolders.cpp

```diff
@@ -1192,11 +1192,35 @@ struct ElideRedundantTransfer : public OpRewritePattern<TensorTransferOp> {
   }
 };
 
+// Attempts to identify trivial case of chained transfer ops (A -> B -> C) and rewrite it as (A -> C)
+// Writes it as A -> B and A -> C relying on dead code elimination to remove the unused A -> B transfer.
+struct ElideChainedTransfer : public OpRewritePattern<TensorTransferOp> {
```

Comment:
transfers are for changing affinity, which needs to be handled here. Add affinities to the test and ensure that the transfer has the target affinity assigned (it doesn't below in the replace op).